server/plugins/irc-events/link.ts

```diff
@@ -488,6 +499,15 @@ function fetch(uri: string, headers: Record<string, string>) {
 	return promise;
 }
 
+function getTransportLayerEncodingLabel(contentType: string): string | undefined {
+	try {
+		const mimeType = new MIMEType(contentType);
+		return (mimeType.parameters.get("charset") as string) || undefined;
```

Comment:
absolutely not, we don't introduce new cast like that.
They are a maintenance hassle and prevent the type checks from working.
Why did you do that? The code was compiling no?

Reply:
> Why did you do that? The code was compiling no?

Yeah, the code was compiling fine, but I got notified about this workflow-run eslint failure:
https://github.com/thelounge/thelounge/actions/runs/12826124441/job/36982976340
> /home/runner/work/thelounge/thelounge/server/plugins/irc-events/link.ts
> 505:3  error  Unsafe return of an `any` typed value  @typescript-eslint/no-unsafe-return

So the update I pushed was a (ham-fisted) attempt to fix that.